File: src/1366-Rank-Teams-by-Votes/1366.js
var rankTeams = function(votes) {
    let n = votes[0].length, cnt = [];
    for (let i = 0; i < 26; i++) {
        cnt[i] = new Array(n + 1).fill(0);
        cnt[i][n] = i;
    }
    
    for (let vote of votes) {
        for (let i = 0; i < n; i++) {
            cnt[vote.charCodeAt(i) - 65][i]--;
        }
    }
    return votes[0].split('').sort((a, b) => {
        for (let i = 0; i <= n; i++) {
            if (cnt[a.charCodeAt() - 65][i] == cnt[b.charCodeAt() - 65][i]) continue;
            return cnt[a.charCodeAt() - 65][i] - cnt[b.charCodeAt() - 65][i];
        }
        return 0;
    }).join('');
};

File: src/1366-Rank-Teams-by-Votes/1366.py
class Solution:
    def rankTeams(self, votes: List[str]) -> str:
        cnt = {v: [0] * len(votes[0]) + [v] for v in votes[0]}
        for vote in votes:
            for i, v in enumerate(vote):
                cnt[v][i] -= 1
        return "".join(sorted(votes[0], key=lambda x: cnt[x]))